### ISSUE TITLE
Character order is not returning in Order of Creation

### ISSUE THREAD

**Issue by SunburnedGoose** [Bungie-net/api#614]
Users could sort their characters by order of creation but this is not working at the moment.

**Comment by vthornheart-bng** [comment]
Sweet, thank you!  I will have to take a look at this, I have a strong suspicion about the janky issue causing this to occur.

**Comment by vthornheart-bng** [comment]
Since it's currently broken anyways, I suppose this is a good time to ask: do you all prefer it being returned in order of creation?

Order of creation makes it consistent and reliable, which I think is a strong benefit.  There are other ways of sorting (most recent etc...) that might have benefits as well however.  I'd be interested in folks' thoughts!  Right now I lean toward order of creation still for the sake of that consistency, but if another order would make most users' lives easier perhaps we could roll with that.

**Comment by nine13tech** [comment]
EMPHATICALLY YES!!!
<img thingy> fry, shut up and take my money</img thingy>

**Comment by vthornheart-bng** [comment]
Emphatically yes to keeping order of creation?

And just to make 100% sure we're all talking about the same thing, we're talking about the ordering of DestinyProfileComponent.characterIds?

**Comment by vthornheart-bng** [comment]
Fixed for the deployment next week!  Yeah... it's tempting to do something other than order of creation, but now that I think about it the consistency feels more important.

**Comment by vthornheart-bng** [comment]
In the meantime if you want a manual workaround, note that characterIds are actually created in ascending order: so lower IDs are, by definition, older than newer IDs.  If you sort by character ID ascending you can emulate what I'll be returning in this fix.

**Comment by JacobCallahan** [comment]
@vthornheart-bng I agree that order of creation is likely the best route forward. Either way, consistency is definitely key! Thanks for the quick fix :100:

**Comment by justrealmilk** [comment]
It’d be cool to add dateCreated tbh. Another data point to show and could be used to order characters but that said since IDs are created in ascending order maybe it’s a little redundant

________________________________

**Comment by ckhicks** [comment]
I generally do the date math for last active, which is almost always what I'm actually looking for, but +1 to having the consistency as a known factor definitely being more important.

**Comment by nine13tech** [comment]
@vthornheart-bng - Yes order of creation!

**Comment by bhollis** [comment]
@vthornheart-bng we're getting reports that this still isn't in the order of character creation.

**Comment by floatingatoll** [comment]
I’ll record another report of this _not_ being fixed anymore as well.

**Comment by vthornheart-bng** [comment]
Can either of you give me an account ID that is experiencing the issue?  We're literally just returning the character IDs in ascending order, which should match the order of character creation as those identifiers are also created in ascending order as people make accounts.  If you've found an example where that's not true, definitely pass it my way so I can investigate!

**Comment by stevennic22** [comment]
I'm currently experiencing this on DIM and Ishtar. Main companion app appears to be working as expected.

Only occurs with Battlenet account. Xbox account is working as expected.

Profile: https://bungie.net/en/Profile/4/4611686018467547676/Schmoops22

**Comment by vthornheart-bng** [comment]
Sweet, thank you!  I'll give it a look!

**Comment by vthornheart-bng** [comment]
I've made a bug to track this request, TFS 718354.

**Comment by vthornheart-bng** [comment]
Ooh, actually we may have a solution by coincidence.  I've got to do some edge case testing.

**Comment by vthornheart-bng** [comment]
Okay, so here's the scoop:

There are edge cases where users can indeed create characters with out-of-order character Ids.  A certain set of specific circumstances have to occur in how they're routed in our backend during and between the creation of those characters for it to occur, but the proof is in the pudding that this happens in practice.

I should be able to get us the order in a reliable way once more, but it's going to require me to go in and make changes to a deeper system.  The soonest window that I'd be allowed to do this will likely be December at this point, but I will keep you posted.

I'm going to leave the fix I've already made as-is for now, as for the majority of players should have a much more consistent experience with the existing fix.  For the remaining players for whom this is happening, they will unfortunately have to wait for this deeper fix.
